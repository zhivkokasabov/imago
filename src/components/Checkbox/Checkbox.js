import React from 'react';

import styles from './Checkbox.module.scss';

import classNames from 'classnames';

export default function Checkbox({ active, className }) {
  return (
    <span className={classNames(styles.checkbox, active ? styles['checkbox--active'] : '', className)}>
      <span />
    </span>
  );
}