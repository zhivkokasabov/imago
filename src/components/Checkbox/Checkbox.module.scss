.checkbox {
  display: block;
  width: 1.25rem;
  height: 1.25rem;
  padding: .125rem;
  border: 1px solid rgb(var(--coolgrey));
  box-sizing: border-box;
  background-color: rgb(var(--white));

  &--active span {
    display: block;
    height: 100%;
    width: 100%;
    background-color: rgb(var(--primary-color));
  }
}