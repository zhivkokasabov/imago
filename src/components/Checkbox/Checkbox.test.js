import React from 'react';
import { render } from '@testing-library/react';

import Checkbox from './Checkbox';

describe('Checkbox component', () => {
  it('should render Checkbox', () => {
    const { asFragment } = render(<Checkbox />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render active Checkbox', () => {
    const { asFragment } = render(<Checkbox active />);

    expect(asFragment()).toMatchSnapshot();
  });
});
