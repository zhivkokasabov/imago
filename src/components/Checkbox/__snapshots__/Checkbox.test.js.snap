// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render Checkbox 1`] = `
<DocumentFragment>
  <span
    class="checkbox"
  >
    <span />
  </span>
</DocumentFragment>
`;

exports[`Checkbox component should render active Checkbox 1`] = `
<DocumentFragment>
  <span
    class="checkbox checkbox--active"
  >
    <span />
  </span>
</DocumentFragment>
`;
